frontend/src/pages/application/index.vue

```diff
@@ -174,13 +176,22 @@ export default {
                 const deviceData = await devicesPromise
                 this.applicationDevices = deviceData?.devices
                 const applicationInstances = await instancesPromise
-
-                if (this.features?.['device-groups']) {
-                    const deviceGroupsPromise = ApplicationApi.getDeviceGroups(applicationId)
-                    const deviceGroupsData = await deviceGroupsPromise
-                    this.deviceGroups = deviceGroupsData?.groups
+                if (this.features?.deviceGroups) {
+                    // DeviceGroups could be disabled for this team. However at this point
+                    // in time we may not have looked up the team object yet, so we cannot be
```

Comment:
Where is the first place the team object gets looked up? I'm also not particularly fond of this pattern...

Reply:
Ultimately it comes down to needing to check `team.type.properties.features.deviceGroups` before deciding whether to do *any* device group related work.

I'm not clear why we're loading deviceGroups at such a high-level of the application stack. The DeviceGroups view reloads them regardless. It would seem cleaner to have the Pipeline views decide for itself what it needs to load - and only if necessary.

Reply:
> I'm not clear why we're loading deviceGroups at such a high-level of the application stack. The DeviceGroups view reloads them regardless. It would seem cleaner to have the Pipeline views decide for itself what it needs to load - and only if necessary.

I dont disagree however this was done to follow suit and not differ from the existing placement of code for all of the instances and devices that are loaded in the same way, same place (with regards to pipelines)

Reply:
Via https://github.com/FlowFuse/flowfuse/pull/3235/commits/2ced020281e69ad692057c68a9e7128d65c5b00d I've reworked the order the application loads things, having discovered we can `await` the setting of the team object before anything else is done. This is much cleaner.